### PR TITLE
Handle the error if the value of org-name in toml is "ballerina"

### LINKED ISSUE #11309: `ballerina build` command gets failed when the org-name of Ballerina.toml file is mentioned as ballerina
**Description:**
`ballerina build` command gets failed when the org-name of `Ballerina.toml` file is mentioned as _ballerina_. This should be failed by giving an proper error message without stopping the build silently.

**Steps to reproduce:**
1. Create a project as `package-sample` and initiate it with `ballerina init`.
2. Update the `Ballerina.toml` file as follows:
```
[project]
org-name = "ballerina"
version = "0.0.1"
```
3. Build the module with `ballerina build sample`

The current output is as follows:
```
Compiling source
    ballerina/sample
```
This should be improved with proper error message.

**Affected Versions:**
Ballerina 0.983.0

**OS, DB, other environment details and versions:**
Ubuntu 16.04 LTS

**Suggested Labels (optional):**
Component/central
Type/Bug

**Suggested Assignees (optional):**
@NatashaWso2

### PR DESCRIPTION
**Pull request by suganyasuven**
Fixes #11309 and https://github.com/ballerina-platform/ballerina-lang/issues/20321

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/ballerinalang/toml/parser/ManifestProcessor.java [hunk 1]
```diff
@@ -170,15 +172,21 @@ public static Manifest parseTomlContentAsStream(InputStream inputStream) throws
...
+        String orgName = manifest.getProject().getOrgName();
+        String version = manifest.getProject().getVersion();
+        if (null == orgName || "".equals(orgName)) {
             throw new TomlException("invalid Ballerina.toml file: cannot find 'org-name' under [project]");
         }
-        
-        if (null == manifest.getProject().getVersion() || "".equals(manifest.getProject().getVersion())) {
+
+        if (!COMPILE_BALLERINA_ORG && ("ballerina".equals(orgName))) {
+            throw new TomlException("invalid Ballerina.toml file: 'org-name' under [project] cannot be 'ballerina'");
```
Comment (hemikak) [inline compiler/ballerina-lang/src/main/java/org/ballerinalang/toml/parser/ManifestProcessor.java]: Can we say `invalid Ballerina.toml file: 'ballerina' is a reserved organization name.`

Comment (suganyasuven) [reply compiler/ballerina-lang/src/main/java/org/ballerinalang/toml/parser/ManifestProcessor.java]: Yes. Changed.

**Comment by warunalakshitha** [comment]
Do we have PR to 1.1.x branch?

File: compiler/ballerina-lang/src/main/java/org/ballerinalang/toml/parser/ManifestProcessor.java [hunk 2]
```diff
@@ -170,15 +172,21 @@ public static Manifest parseTomlContentAsStream(InputStream inputStream) throws
...
-        if (null == manifest.getProject().getOrgName() || "".equals(manifest.getProject().getOrgName())) {
+        String orgName = manifest.getProject().getOrgName();
+        String version = manifest.getProject().getVersion();
+        if (null == orgName || "".equals(orgName)) {
             throw new TomlException("invalid Ballerina.toml file: cannot find 'org-name' under [project]");
         }
-        
-        if (null == manifest.getProject().getVersion() || "".equals(manifest.getProject().getVersion())) {
+
+        if (!COMPILE_BALLERINA_ORG && ("ballerina".equals(orgName))) {
```
Comment (warunalakshitha) [inline compiler/ballerina-lang/src/main/java/org/ballerinalang/toml/parser/ManifestProcessor.java]: Shall we add constant to Ballerina. There should be already defined constants.

**Comment by hevayo** [comment]
Closing the PR since we should not special case ballerina build

**Pull request closed without merging**